core/src/main/java/com/linecorp/armeria/common/util/DomainSocketAddress.java

```diff
@@ -15,28 +15,36 @@
  */
 package com.linecorp.armeria.common.util;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.net.URI;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Set;
 
 import com.linecorp.armeria.client.Endpoint;
 import com.linecorp.armeria.common.annotation.Nullable;
 import com.linecorp.armeria.common.annotation.UnstableApi;
 import com.linecorp.armeria.internal.common.util.DomainSocketUtil;
 
 /**
- * An {@link InetSocketAddress} that refers to the {@link Path} of a Unix domain socket.
+ * An {@link InetSocketAddress} that refers to a Unix domain socket path.
  * This class extends {@link InetSocketAddress} to ensure the backward compatibility with existing
  * Armeria API that uses {@link InetSocketAddress} in its API. This address will have the following properties:
  * <ul>
  *   <li>hostname and hostString - {@code "unix:<path>"}, e.g. {@code "unix:/var/run/server.sock"}</li>
  *   <li>address - an IPv6 address that falls into <a href="https://datatracker.ietf.org/doc/rfc6666/">IPv6
  *       Discard Prefix</a></li>
  * </ul>
+ *
+ * <h2>Pitfalls</h2>
+ * <p>Comparing two {@link DomainSocketAddress}es using {@link #equals(Object)} will always return {@code true}
+ * because there's no way to override {@link InetSocketAddress#equals(Object)}, which compares only the IP
+ * addresses for resolved addresses. You should never use {@link DomainSocketAddress} as a key of a {@link Map},
+ * or as an element of a {@link Set}; consider using {@link Endpoint} instead.</p>
```

Comment:
Optional) We may implement `Comparable<T>` and suggest users should use `TreeMap` if necessary. However, the current warning seems sufficient.

Reply:
I'm not sure if it's a good idea to implement `Comparable` here given we can't change `equals()` and `hashCode()` anyway. Let me leave this as it is for now.